File: exercises-python/2. Problemas de estructura condicional/38.py
'''
38.Desarrolle el programa que determine la cantidad de días
y el nombre del mes conociendo los valores numéricos del mes
y del año. Un año es bisiesto si es divisible por 4 y no es divisible
por 100, o no es divisible por 400.
'''
import os
os.system("cls")

mes = int(input("Mes: "))
ano = int(input("Año: "))
mesDias = 0
if mes == 1:
    nombreMes = "Enero"
    mesDias = 31
elif mes == 2:
    nombreMes = "Febrero"
    if ano % 4 == 0 and ano % 100 != 0:
        mesDias = 29
    else:
        mesDias = 28
elif mes == 3:
    nombreMes = "Marzo"
    mesDias = 31
elif mes == 4:
    nombreMes = "Abril"
    mesDias = 30
elif mes == 5:
    nombreMes = "Mayo"
    mesDias = 31
elif mes == 6:
    nombreMes = "Junio"
    mesDias = 30
elif mes == 7:
    nombreMes = "Julio"
    mesDias = 31
elif mes == 8:
    nombreMes = "Agosto"
    mesDias = 31
elif mes == 9:
    nombreMes = "Septiembre"
    mesDias = 30
elif mes == 10:
    nombreMes = "Octubre"
    mesDias = 31
elif mes == 11:
    nombreMes = "Noviembre"
    mesDias = 30
elif mes == 12:
    nombreMes = "Diciembre"
    mesDias = 31
else: print("Error de número de mes.")

if mesDias != 0:
    print(f"Días del mes: {mesDias}")

File: exercises-cpp/2. Condicionales/38.cpp
 #include <iostream>
#include <cstdio>
#include <strings.h>
#include <conio.h>
using namespace std;

int main(){
    system("cls");

    int mes,ano,mesDias;
    string nombreMes;

    printf("Mes: ");cin>>mes;
    printf("Año: ");cin>>ano;

    if (mes == 1){
        nombreMes = "Enero";
        mesDias = 31;
        }
    else if (mes == 2){
        nombreMes = "Febrero";
        if (ano % 4 == 0 and ano % 100 != 0){
            mesDias = 29;
        }
        else{
            mesDias = 28;
        }
    }
    else if (mes == 3){
        nombreMes = "Marzo";
        mesDias = 31;
    }
    else if (mes == 4){
        nombreMes = "Abril";
        mesDias = 30;
    }
    else if (mes == 5){
        nombreMes = "Mayo";
        mesDias = 31;
    }
    else if (mes == 6){
        nombreMes = "Junio";
        mesDias = 30;
    }
    else if (mes == 7){
        nombreMes = "Julio";
        mesDias = 31;
    }
    else if (mes == 8){
        nombreMes = "Agosto";
        mesDias = 31;
    }
    else if (mes == 9){
        nombreMes = "Septiembre";
        mesDias = 30;
    }
    else if (mes == 10){
        nombreMes = "Octubre";
        mesDias = 31;
    }
    else if (mes == 11){
        nombreMes = "Noviembre";
        mesDias = 30;
    }
    else if (mes == 12){
        nombreMes = "Diciembre";
        mesDias = 31;
    }
    else printf("Error de número de mes.");

    if (mesDias != 0) {
        cout<<"Días del mes: "<<mesDias<<endl;
    }
}